### PR TITLE
move away from custom wait for compaction to using ccm implementation

### PR DESCRIPTION
**Pull request by knifewine**

### CODE REVIEW

**Comment by knifewine** [comment]
tested relevant tests locally against trunk and 2.1.

@mambocab can you review?

File: compaction_test.py [hunk 1]
```diff
@@ -172,7 +174,9 @@ def sstable_deletion_test(self):
         for x in range(0, 100):
             session.execute('delete from cf where key = ' + str(x))
 
-        block_on_compaction_log(node1, ks='ks', table='cf')
+        node1.flush()
+        node1.compact()
```
Comment (mambocab) [inline compaction_test.py]: This changes behavior a bit -- the new implementation compacts everything, not just `ks.cf`. Is that intentional? Is preserving that behavior important?

**Comment by mambocab** [comment]
Running locally, but code looks good -- feel free to merge after my comment is addressed.

File: compaction_test.py (hunk 1 above)
Comment (knifewine) [inline compaction_test.py]: Yeah, was intentional, as far as I could see the tests behave fine while doing a full compaction, so didn't see the need to keep the complexity.

ccm's Node.compact() implementation doesn't currently allow setting specific ks/table, though we could update that or call directly to Node.nodetool with custom params.

IMO we're fine to just do a wholesale compaction for now but can change it up too.

Comment (mambocab) [inline compaction_test.py]: Gotcha -- yeah, sounds good to me. I don't imagine there's enough data in other tables to make compaction take noticably longer across everything, yeah?

> ccm's Node.compact() implementation doesn't currently allow setting specific ks/table

Side note: that's surprising. I'll file a ccm ticket.

Comment (knifewine) [inline compaction_test.py]: Actually, on second thought it's trivial to just use nodetool here, so I'll do that and preserve compacting only ks.cf

Tested and it works just fine.

**Comment by knifewine** [comment]
thanks for the feedback, fixed it up to keep behavior the same as before. will squash and merge once travis is happy.

**Comment by mambocab** [comment]
:+1:

**Pull request merged**